oval_graph/_client_create_html_graph.py

```diff
@@ -0,0 +1,126 @@
+    self.open_web_browser(src)
+
+
+def open_web_browser(self, src):
+    if not self.off_webbrowser:
+        try:
+            webbrowser.get('firefox').open_new_tab(src)
+        except BaseException:
+            webbrowser.open_new_tab(src)
+
+
+def get_src(self, src):
+    _dir = os.path.dirname(os.path.realpath(__file__))
+    FIXTURE_DIR = os.path.join(_dir, src)
+    return str(FIXTURE_DIR)
+
+
+def _prepare_data(self, rules, dict_oval_trees, out, date):
+    for rule in rules['rules']:
+        try:
+            self._put_to_dict_oval_trees(dict_oval_trees, rule)
+            if not self.all_in_one:
+                src = self._get_src_for_one_graph(rule, date)
+                self.save_html_and_open_html(
+                    dict_oval_trees, src, dict(
+                        rules=[rule]), out)
+                dict_oval_trees = {}
+        except NotChecked as error:
+            self.print_red_text(error)
+    if self.all_in_one:
+        src = self.get_save_src('rules' + date)
+        self.save_html_and_open_html(dict_oval_trees, src, rules, out)
+    return out
+
+
+def prepare_data(self, rules):
+    out = []
+    oval_tree_dict = dict()
+    date = str(datetime.now().strftime("-%d_%m_%Y-%H_%M_%S"))
+    out = self._prepare_data(rules, oval_tree_dict, out, date)
+    return out
+
+
+def get_save_src(self, rule):
+    if self.out is not None:
+        os.makedirs(self.out, exist_ok=True)
+        return os.path.join(self.out, 'graph-of-' + rule + '.html')
+    return os.path.join(os.getcwd(), 'graph-of-' + rule + '.html')
+
+
+def _get_part(self, part):
+    with open(os.path.join(self.parts, part), "r") as data_file:
+        return data_file.readlines()
+
+
+def _merge_report_parts(self, data):
+    head = self._get_part('head.txt')
+    css = self._get_part('css.txt')
+    boot_strap_style = self._get_part('bootstrapStyle.txt')
+    jsTree_style = self._get_part('jsTreeStyle.txt')
+    jQuery_script = self._get_part('jQueryScript.txt')
+    boot_strap_script = self._get_part('bootstrapScript.txt')
+    jsTree_script = self._get_part('jsTreeScript.txt')
+    body_start = ['</head>', '<body>', data]
+    body = self._get_part('body.txt')
+    script = self._get_part('script.js')
+    footer = ['<script>', *script, '</script>', '</body>', '</html>']
+    return [
+        *head,
+        *css,
+        *boot_strap_style,
+        *jsTree_style,
+        *jQuery_script,
+        *boot_strap_script,
+        *jsTree_script,
+        *body_start,
+        *body,
+        *footer]
+
+
+def save_html_report(self, dict_, src):
+    data = (
+        "\n<script>var data_of_tree = " + str(
+            json.dumps(
+                {
+                    re.sub(
+                        r'\-\d{2}\_\d{2}\_\d{4}\-\d{2}\_\d{2}\_\d{2}$',
+                        '',
+                        k): v for k,
+                    v in dict_.items()},
+                sort_keys=False,
+                indent=4)) + ";</script>\n")
```

Comment:
This is kind of complex statement. Try to simplify it. Maybe start by extracting the code inside `str(...)` to a new variable.